### PR TITLE
Handle broadcast to invalid cycle points and namespaces.

### LINKED ISSUE #1415: Bad use of broadcast can bring down a suite.
E.g. in a date-time cycling suite:
```
% cylc broadcast -s 'command scripting=true' -p 1 -n foo-1 foo
# ... (suite shuts down)
% cylc cat-log -e foo | tail -n 31
Traceback (most recent call last):
  File "/home/oliverh/cylc/cylc.git/lib/cylc/run.py", line 75, in main
    server.run()
  File "/home/oliverh/cylc/cylc.git/lib/cylc/scheduler.py", line 908, in run
    self.pool.wireless.expire( self.pool.get_min_point() )
  File "/home/oliverh/cylc/cylc.git/lib/cylc/broadcast.py", line 139, in expire
    get_point(point_string) < cutoff_point):
  File "/home/oliverh/cylc/cylc.git/lib/cylc/cycling/iso8601.py", line 116, in __cmp__
    return self._iso_point_cmp(self.value, other.value)
  File "/home/oliverh/cylc/cylc.git/lib/cylc/cycling/iso8601.py", line 85, in _wrapper
    results = function(*args)
  File "/home/oliverh/cylc/cylc.git/lib/cylc/cycling/iso8601.py", line 154, in _iso_point_cmp
    point = point_parse(point_string)
  File "/home/oliverh/cylc/cylc.git/lib/cylc/cycling/iso8601.py", line 695, in point_parse
    return _point_parse(point_string).copy()
  File "/home/oliverh/cylc/cylc.git/lib/cylc/cycling/iso8601.py", line 85, in _wrapper
    results = function(*args)
  File "/home/oliverh/cylc/cylc.git/lib/cylc/cycling/iso8601.py", line 715, in _point_parse
    point = SuiteSpecifics.point_parser.parse(point_string)  # Fail?
  File "/home/oliverh/cylc/cylc.git/lib/isodatetime/parsers.py", line 236, in parse
    date_info, time_info, parsed_expr = self.get_info(timepoint_string)
  File "/home/oliverh/cylc/cylc.git/lib/isodatetime/parsers.py", line 432, in get_info
    keys, date_info = self.get_date_info(date)
  File "/home/oliverh/cylc/cylc.git/lib/isodatetime/parsers.py", line 392, in get_date_info
    raise ISO8601SyntaxError("date", date_string)
ISO8601SyntaxError: Invalid ISO 8601 date representation: 1
ERROR CAUGHT: cleaning up before exit
2015-04-15T22:12:23+12 WARNING - some active tasks will be orphaned
THE ERROR WAS:
Invalid ISO 8601 date representation: 1
use --debug to turn on exception tracebacks)
```

### PR DESCRIPTION
**Pull request by hjoliver**
Fix #1415.

### CODE REVIEW

**Comment by hjoliver** [comment]
@matthewrmshin - please review.

File: tests/broadcast/05-bad-point/suite.rc [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+[runtime]
+    [[broadcast]]
+        command scripting="""
+# Broadcast to an integer point, not valid for this suite; and
+# fail if the broadcast succeeds (it should fail).
+if cylc broadcast -s 'title=foo' -p '1' "${CYLC_SUITE_NAME}"; then
+   /bin/false
+else
+   /bin/true
+fi"""
```
Comment (matthewrmshin) [inline tests/broadcast/05-bad-point/suite.rc]: Why isn't this just:

```sh
! cylc broadcast -s 'title=foo' -p '1' "${CYLC_SUITE_NAME}"
```

**Comment by matthewrmshin** [comment]
I am happy with this change regarding the functionality. The minor comment for the test does not affect functionality of the system, so I am happy for this change to merge as-is.

File: tests/broadcast/05-bad-point/suite.rc (hunk 1 above)
Comment (hjoliver) [inline tests/broadcast/05-bad-point/suite.rc]: oops, yes - I originally had echo statements in those code blocks.

**Comment by hjoliver** [comment]
(ready to merge)

**Pull request merged**